### PR TITLE
[inductor] Add --inductor-config benchmark flag

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129036
* #129035
* #127506
* #129026
* #129025
* #126417
* __->__ #129034
* #129033
* #129024

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -4025,6 +4031,18 @@ def run(runner, args, original_dir=None):
         inductor_config.triton.divisible_by_16 = not args.disable_divisible_by_16
         if args.inference:
             inductor_config.freezing = args.freezing
+        if args.inductor_config:
+            for config in args.inductor_config:
+                key, value = config.split("=")
+                typ = type(inductor_config.__getattr__(key))
```
Comment (shunting314) [inline benchmarks/dynamo/common.py:4037]: Do we want to support config like inductor_config.triton.foo . We need repeatedly call getattr to handle them

**Review by shunting314 — APPROVED** [review]

**Review by eellison — APPROVED** [review]

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (jansel) [reply benchmarks/dynamo/common.py:4037]: This already supports that.  My use case is `-c halide.scheduler_cpu=...`.

inductor_config is secretly just a dict, the recursive objects is just syntactic sugar.

**Comment by jansel** [comment]
pytorchbot merge

**Pull request closed without merging**